dcpy/lifecycle/package/assemble.py

```diff
@@ -141,6 +144,12 @@ def pull_destination_files(
     package_ids = {p.id for p in product_metadata.assembly}
     for f in dest.files:
         paths_and_dests = ids_to_paths_and_dests[f.id]
+        f_is_metadata = (
+            f.id in product_metadata.get_file_ids()
+            and product_metadata.get_file_and_overrides(f.id).file.is_metadata
+        )
+        if metadata_only and not f_is_metadata:
+            continue
```

Comment:
Codecov flagging - this might actually be a worthwhile test (not so much for this pr specifically, but just ensure that metadata_only updates only update metadata

Reply:
very good call! added.